compiler/rustc_mir_build/src/thir/cx/expr.rs

```diff
@@ -130,6 +130,7 @@ impl<'tcx> Cx<'tcx> {
                 ExprKind::Pointer { cast: PointerCast::Unsize, source: self.thir.exprs.push(expr) }
             }
             Adjust::Pointer(cast) => ExprKind::Pointer { cast, source: self.thir.exprs.push(expr) },
+            Adjust::NeverToAny if adjustment.target.is_never() => return expr,
```

Comment:
is this testable in a mir-opt test without `#[custom_mir]`?

Reply:
It looks like it is: https://github.com/rust-lang/rust/pull/111501#issuecomment-1545543420 😅